src/main/java/io/github/minecraftcursedlegacy/api/client/AtlasMap.java

```diff
@@ -0,0 +1,169 @@
+	}
+
+	/**
+	 * Register the given atlas for use with the {@link ItemType} corresponding to the given ID
+	 * when {@link ItemInstance#getDamage()} on an instance of it is the same as the given meta.
+	 *
+	 * @param itemID The ID of the item which uses the given atlas
+	 * @param meta The damage value an instance must have to use this atlas
+	 * @param atlas The location of the custom atlas
+	 *
+	 * @throws NullPointerException If atlas is <tt>null</tt>
+	 */
+	public static void registerAtlas(int itemID, int meta, String atlas) {
+		AtlasMapper.registerAtlas(itemID, meta, atlas);
+	}
+
+	/**
+	 * Register the given atlas for use with the {@link ItemType} corresponding to the given ID,
+	 * regardless of what {@link ItemInstance#getDamage()} on an instance of it returns.
+	 *
+	 * @param itemID The ID of the item which uses the given atlas
+	 * @param atlas The location of the custom atlas
+	 *
+	 * @throws NullPointerException If atlas is <tt>null</tt>
+	 */
+	public static void registerAtlas(int itemID, String atlas) {
+		AtlasMapper.registerDefaultAtlas(itemID, atlas);
+	}
+
+	/**
+	 * Register the given sprite for use with the {@link ItemType} corresponding to the given 
+	 * {@link ItemInstance#itemId} when it has the given {@link ItemInstance#getDamage()}.
+	 *
+	 * @param stack The item type and damage which uses the given sprite
+	 * @param sprite The location of the custom sprite
+	 *
+	 * @return The atlas positioning for use with {@link ItemType#method_458(int)}
+	 *
+	 * @throws NullPointerException If sprite is <tt>null</tt>
+	 * @throws IllegalArgumentException If there is already an atlas or sprite registered for the given ID and meta
+	 */
+	public static int registerSprite(ItemInstance stack, String sprite) {
+		return registerSprite(stack.itemId, stack.getDamage(), sprite);
+	}
+
+	/**
+	 * Register the given sprite for use with the given {@link ItemType},
+	 * regardless of what {@link ItemInstance#getDamage()} on an instance of it returns.
+	 *
+	 * @param item The item which uses the given sprite
+	 * @param sprite The location of the custom sprite
+	 *
+	 * @return The atlas positioning for use with {@link ItemType#method_458(int)}
+	 *
+	 * @throws NullPointerException If sprite is <tt>null</tt>
+	 * @throws IllegalArgumentException If there is already an atlas or sprite registered for the given ID
+	 */
+	public static int registerSprite(ItemType item, String sprite) {
+		return registerSprite(item.id, sprite);
+	}
+
+	/**
+	 * Register the given sprite for use with the {@link ItemType} corresponding to the given ID
+	 * when {@link ItemInstance#getDamage()} on an instance of it is the same as the given meta.
+	 *
+	 * @param itemID The ID of the item which uses the given sprite
+	 * @param meta The damage value an instance must have to use this sprite
+	 * @param sprite The location of the custom sprite
+	 *
+	 * @return The atlas positioning for use with {@link ItemType#method_458(int)}
```

Comment:
Do you have a name in mind for mapping `ItemType#method_458`?

Reply:
Well in `ItemType` you've got two methods for setting the atlas position:
```java
public ItemType method_458(int index) {
	this.texturePosition = index;
	return this;
}

public ItemType setTexturePosition(int x, int y) {
	this.texturePosition = x + y * 16;
	return this;
}
```
At a simplest level they could both be `setTexturePosition` and the coordinate one overloads the index one. It might be better if the index one was `setTexturePosition` and the coordinate one rename to `setTextureCoordinates` or something along those lines to differentiate.